cpp/src/datetime/datetime_ops.cu

```diff
@@ -54,7 +54,11 @@ enum class datetime_component {
   NANOSECOND
 };
 
-enum class rounding_kind { CEIL, FLOOR };
+enum class rounding_kind {
+  CEIL,   ///< Rounds up to the next integer multiple of the provided frequency
+  FLOOR,  ///< Rounds down to the next integer multiple of the provided frequency
+  ROUND   ///< Rounds to the nearest integer multiple of the provided frequency
```

Comment:
`rounding_kind::ROUND` isn't very descriptive. How about "nearest"?

Reply:
We might even want a name like `NEAREST_HALF_UP` since other parts of cuDF have support for rounding halves to the nearest even value (which this does not need to handle, to my knowledge). https://github.com/rapidsai/cudf/blob/8002cbd87367135a941d1145c9d489a8f82dc76d/cpp/include/cudf/round.hpp#L30-L36

Reply:
On the other hand, keeping the name "round" emphasizes the alignment with "ceil" and "floor", all of which have corresponding function names in pandas and `std::chrono`.

Reply:
I like the idea of renaming `rounding_kind` to `rounding_function` if that conveys a better message and if it solves the problem. Let me know!

Reply:
@jrhemstad Do you have a preference among these suggestions?

Reply:
`rounding_function` sounds good

Reply:
@bdice does adding this, make this PR a breaking change? Do I need to change any labels?

Reply:
Nope, this enum is in the `cudf::detail` namespace. Changes to private or detail APIs do not require a breaking label.